src/control/lib/spdk/nvme_test.go

```diff
@@ -137,6 +141,121 @@ func TestSpdk_cleanLockfiles(t *testing.T) {
 	}
 }
 
+func TestSpdk_cleanKnownLockfiles(t *testing.T) {
+	testDir, clean := test.CreateTestDir(t)
+	defer clean()
+
+	for name, tc := range map[string]struct {
+		addrsToClean []string
+		lfAddrsInDir []string
+		expRemoved   []string
+		expRemaining []string
+		expErr       error
+	}{
+		"no lockflles; no addresses to clean": {
+			expRemoved:   []string{},
+			expRemaining: []string{},
+		},
+		"no lockfiles": {
+			addrsToClean: []string{
+				"0000:81:00.0",
+				"0000:82:00.0",
+				"0000:83:00.0",
+			},
+			lfAddrsInDir: []string{},
+			expRemoved:   []string{},
+			expRemaining: []string{},
+		},
+		"no addresses to clean": {
+			lfAddrsInDir: []string{
+				"0000:81:00.0",
+				"0000:82:00.0",
+				"0000:83:00.0",
+			},
+			addrsToClean: []string{},
+			expRemoved:   []string{},
+			expRemaining: []string{
+				LockfilePrefix + "0000:81:00.0",
+				LockfilePrefix + "0000:82:00.0",
+				LockfilePrefix + "0000:83:00.0",
+			},
+		},
+		"clean some addresses": {
+			lfAddrsInDir: []string{
+				"0000:81:00.0",
+				"0000:82:00.0",
+				"0000:83:00.0",
+			},
+			addrsToClean: []string{
+				"0000:82:00.0",
+				"0000:83:00.0",
+			},
+			expRemoved: []string{
+				filepath.Join(testDir, LockfilePrefix+"0000:82:00.0"),
+				filepath.Join(testDir, LockfilePrefix+"0000:83:00.0"),
+			},
+			expRemaining: []string{
+				LockfilePrefix + "0000:81:00.0",
+			},
+		},
+		"clean all addresses": {
+			lfAddrsInDir: []string{
+				"0000:81:00.0",
+				"0000:82:00.0",
+				"0000:83:00.0",
+			},
+			addrsToClean: []string{
+				"0000:84:00.0",
+				"0000:82:00.0",
+				"0000:83:00.0",
+				"0000:81:00.0",
+			},
+			expRemoved: []string{
+				filepath.Join(testDir, LockfilePrefix+"0000:81:00.0"),
+				filepath.Join(testDir, LockfilePrefix+"0000:82:00.0"),
+				filepath.Join(testDir, LockfilePrefix+"0000:83:00.0"),
+			},
+			expRemaining: []string{},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			if err := test.RemoveContents(t, testDir); err != nil {
+				t.Fatal(err)
+			}
+
+			mnc := MockNvmeCfg{
+				LockfileDir: testDir,
+				RemoveFn:    os.Remove,
+			}
+			mockImpl := MockNvmeImpl{Cfg: mnc}
```

Comment:
Thanks for adding this test. But why mock here? Testing the real NvmeImpl makes more sense for this one.